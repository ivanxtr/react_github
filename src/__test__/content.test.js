import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Content from '../components/content';

Enzyme.configure({ adapter: new Adapter() });

const mockData = [
  {
    "title": "one"
  },
  {
    "title": "two"
  },
  {
    "title": "three"
  }
]

describe('Conten Component', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Content />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot render', () => {
    const component = create(<Content />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render the proper content', () => {
    const wrapper = mount(<Content data={mockData} />)
    expect(wrapper.find('.bugs')).toHaveLength(1);
    expect(wrapper.find('.item')).toHaveLength(3);
    expect(wrapper.find('.item').at(0).text()).toBe("one");
    expect(wrapper.find('.item').at(1).text()).toBe("two");
    expect(wrapper.find('.item').at(2).text()).toBe("three");
  })
})
