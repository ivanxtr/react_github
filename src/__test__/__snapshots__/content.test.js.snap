// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conten Component snapshot render 1`] = `
<div
  className="bugs"
>
  <div>
    No data to display
  </div>
</div>
`;
